### ISSUE TITLE
FNX2-13576 ⁃ [Bug] Crash when enabling reader mode while the page is being scrolled

### ISSUE THREAD

**Issue by Diana-Rus** [mozilla-mobile/fenix#11696]
## Steps to reproduce
1. Open app and go to wikipedia.org/wiki/mozilla
1. Scroll the page.
1. Tap on the reader mode icon.
    1. Expected behavior
Reader mode for the page is enabled.
    1. Actual behavior
Crash is present.
    1. Device information

* Android device: Samsung Galaxy S9 (Android 8)
* Fenix version: Beta 5.2.0-beta.2, Nightly 6/17

Crash ID: 16dc8b06-1e7b-49f0-9b13-33b950200617
Crash signature: mozilla::a11y::SessionAccessibility::SendScrollingEvent
Socorro: https://crash-stats.mozilla.org/report/index/bp-16dc8b06-1e7b-49f0-9b13-33b950200617

GIF
![20200617-192131](https://user-images.githubusercontent.com/49859863/84923957-98b66d80-b0d0-11ea-9a7f-f92a453e348a.gif)

Note:

* Seems that it's about timing as well as it can be reproducible when the scrolling, slows down and right about then tapping on the reader mode icon.
* I didn't encounter the crash with Google Pixel 3 XL (Android 9) neither with Sony Xperia Z5 (Android 7) but instead of crashing, the icon for reader mode is colored as when it's active, even if it's not the case(reader mode is not applied).
![Screenshot](https://user-images.githubusercontent.com/49859863/84925440-90f7c880-b0d2-11ea-857a-57e2c69fff2e.png)UNITO-UNDERSCORE!20200617-194238!
* For step 1 you can also use an article from nytimes.com and same outcome is present

**Comment by csadilek** [comment]
Thanks! Can't reproduce on any of my devices either, but filed https://bugzilla.mozilla.org/show_bug.cgi?id=1646437 for the native crash in a11y.

**Comment by Gaspacchio** [comment]
Can reproduce on my Samsung S8, but on this page: https://www.techradar.com/how-to/facebook-dark-mode

## Device information

- Android device: Samsung Galaxy S8 (Sm-G950F) (Android 9)
- Fenix version: 5.1.1 (Build #21571852)

Crash ID: `76dbc4bd-c799-48f1-bf54-a2d740200618`
Crash signature: `mozilla::a11y::DocAccessiblePlatformExtParent::RecvSetPivotBoundaries`
Socorro: https://crash-stats.mozilla.org/report/index/76dbc4bd-c799-48f1-bf54-a2d740200618

I know my report doesn't have the same signature, tell me if you need me to open a new issue 🙏!

**Comment by csadilek** [comment]
@Gaspacchio Thanks, no need to file, we already have it: https://bugzilla.mozilla.org/show_bug.cgi?id=1635722

Both bugs appear to be fixed and will be in next Nightly. Let's verify once available.

**Comment by Diana-Rus** [comment]
Hey @Gaspacchio , one questions i can only trigger the crash mentioned in description. Did you do anything else in order to encounter the crash that you mentioned?
Thanks 😊

**Comment by Gaspacchio** [comment]
No, nothing particular @Diana-Rus

**Comment by csadilek** [comment]
@Diana-Rus According to https://bugzilla.mozilla.org/show_bug.cgi?id=1646437, this should be fixed in Nightly now. Can you verify, please? :)

**Comment by Gaspacchio** [comment]
Bug fixed for me, thanks! 🎉

**Comment by Diana-Rus** [comment]
Hi, verified the crash as fixed on Nightly 06/30 with:
- Samsung Galaxy S9 (Android 8)
- Google Pixel 3XL (Android 9)
Samsung - GIF
![20200630-164014](https://user-images.githubusercontent.com/49859863/86133531-175bd380-baf1-11ea-9c8f-00f6d7287ffe.gif)

Note:
- @csadilek  While testing however i encountered a crash with the following signature libxul.so@0x1b8ba52 | anon_27155_27197 (deleted)@0x5a17e and URL: https://crash-stats.mozilla.org/report/index/2efda222-279c-4cad-ad7c-fcc3a0200630
I managed to reproduce it after many attempts(10 to 13 times) just by enabling and disabling the reader mode while scrolling

**Comment by Diana-Rus** [comment]
As an update, the crash mentioned above is on Samsung Galaxy S9 (Android 8).
When trying to replicate the issue on Pixel there is a different crash present, but still the signature contains similarity with previous encountered -> libxul.so@0x1b8ba52

Encountered Signatures on Pixel 3 XL:
1.   libxul.so@0x1b8ba52 | libart.so@0x23326d | libxul.so@0x1b8bc69 | libxul.so@0x1b8b7c7 | libxul.so@0x1b8b759 | libxul.so@0xb7a277 | libxul.so@0x2d73917 | libxul.so@0x29405bf | libxul.so@0x47dc136 | libart.so@0x3731d9 | libart.so@0x195ffe | dalvik-main sp... 
Crash URL: https://crash-stats.mozilla.org/report/index/ceae94a3-8f6f-4d12-8be4-faf3d0200630
2.  libxul.so@0x1b8ba52 | libart.so@0x23326d | libxul.so@0x1b8bc69 | libxul.so@0x1b8b7c7 | libxul.so@0x1b8b759 | libxul.so@0xb7a277 | libxul.so@0x2d73917 | dalvik-main space (region space) (deleted)@0x5f9c2e
Crash URL: https://crash-stats.mozilla.org/report/index/6927a4f6-8d5a-4783-9147-45e570200630
3 . libxul.so@0x1b8ba52 | libart.so@0x23326d | libxul.so@0x1b8bc69 | libxul.so@0x1b8b7c7 | libxul.so@0x1b8b759 | libxul.so@0xb7a277 | libxul.so@0x2d73917 | libxul.so@0x29405bf | libxul.so@0x47dc136 | libxul.so@0xaebffd | libart.so@0x2d37e9 | dalvik-main sp...

**Note: tried to narrow it to a more specific reproduction - it seems to be happening when tapping while the reader mode icon is not displayed in the URL bar and then tapping where the reader mode icon should be.**

**Comment by liuche** [comment]
The original STR is verified, but let's file this as a separate bug and close this one as verified.

**Comment by kbrosnan** [comment]
We need symbols for the Nightly builds before we can say that they are different crashes.

**Comment by liuche** [comment]
@kbrosnan is there a bug tracking the Nightly symbols work, so we can check and figure out when this can be re-checked.

**Comment by kbrosnan** [comment]
This is the only place I have encountered unsymbolicated crashes. Closing as verified from https://github.com/mozilla-mobile/fenix/issues/11696#issuecomment-651809572
